src/Database/Esqueleto/Record.hs

```diff
@@ -115,8 +119,51 @@ import Data.Maybe (mapMaybe, fromMaybe, listToMaybe)
 --
 -- @since 3.5.6.0
 deriveEsqueletoRecord :: Name -> Q [Dec]
-deriveEsqueletoRecord originalName = do
-  info <- getRecordInfo originalName
+deriveEsqueletoRecord = deriveEsqueletoRecordWith defaultSettings
+
+-- | Codegen settings for 'deriveEsqueletoRecordWith'.
+--
+-- @since 3.5.8.0
+data Settings = Settings
```

Comment:
The `Settings` name is going to be a really common conflict. Let's give it a more specific name:

```suggestion
data DeriveEsqueletoRecordSettings = DeriveEsqueletoRecordSettings
```

Reply:
If you prefer. I tend to think of the module name as disambiguating these kinds of general names.